src/compiler/checker.ts

```diff
@@ -38913,7 +38913,9 @@ namespace ts {
             }
 
             if (!container) {
-                grammarErrorOnFirstToken(node, Diagnostics.A_return_statement_can_only_be_used_within_a_function_body);
+                if (!!getSourceFileOfNode(node).externalModuleIndicator || !isInJSFile(node)) {

```

Comment:
```suggestion
                if (!isInJSFile(node)) {
```

Reply:
In fact, I would say that the right fix is still to issue the error in checkJs, but not when providing the default grammar errors for JS files.